Mobile Buy SDK/Mobile Buy SDK/Additions/NSDateFormatter+BUYAdditions.m

```diff
@@ -38,8 +38,29 @@ + (NSDateFormatter*)dateFormatterForShippingRates
 + (NSDateFormatter*)dateFormatterForPublications
 {
 	NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+	
+	if (![self is24hTimeFormat]) {
+		[dateFormatter setLocale:[NSLocale localeWithLocaleIdentifier:@"en_GB"]];
+	}
+	
 	dateFormatter.dateFormat = @"yyyy-MM-dd'T'HH:mm:ssZ";
 	return dateFormatter;
 }
 
++ (BOOL)is24hTimeFormat
+{
+	NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
+	[formatter setLocale:[NSLocale currentLocale]];
+	[formatter setDateStyle:NSDateFormatterNoStyle];
+	[formatter setTimeStyle:NSDateFormatterShortStyle];
+	
+	NSString *dateString = [formatter stringFromDate:[NSDate date]];
+	
+	NSRange amRange = [dateString rangeOfString:[formatter AMSymbol]];
+	NSRange pmRange = [dateString rangeOfString:[formatter PMSymbol]];
+	BOOL is24h = (amRange.location == NSNotFound && pmRange.location == NSNotFound);
+	
+	return is24h;
```

Comment:
Probably it is redundant to change server api for this case. I will explain. Date coming from server as "2017-05-31T14:57:48+01:00" (for example). In this format possible hours value is 00-24. In general dateformatter is working ok with it, but for some locales and 12-hour time mode it can't parse dates with hours more that 11 (example "2017-05-31T11:57:48+01:00" will instantiate proper date, "2017-05-31T14:57:48+01:00" will return nil date object).

Steps to reproduce it on your device:
1. In Settings->General->Language & Region->Region set United Kingdom
2. In time Settings->General->Date & Time set 24-Hour Time off
3. Run the app and check any dates for product in SDK. The dates where HH > 11 will not be instantiated.

Solution. NSDate is in fact Unix timestamp, and "2017-05-31T11:57:48+01:00" includes UTC timezone, so there are no difference what locale you will set to dateFormatter to instantiate NSDate object - it will be the same date. So to avoid #665 issue on affected devices you can just set any locale for NSDateFormatter, instead user's current. 

I did some more investigations of SDK. + (NSDateFormatter*)dateFormatterForPublications is used to instantiate string from date, so users locale should be saved. Check this code, just re-instantiating current user's locale with the same locale solve the problem - dates instantiating normally on affected devices:
NSLocale *currentLocale = [dateFormatter locale];
NSString *identifier = currentLocale.localeIdentifier;

if (![self is24hTimeFormat]) {
	[dateFormatter setLocale:[NSLocale localeWithLocaleIdentifier:identifier]];
}